src/tree_cmd.cpp

```diff
@@ -829,15 +829,29 @@ bool DecrementTreeCounter()
 	return old_trees_tick_ctr <= _trees_tick_ctr;
 }
 
+/**
+ * Place a random tree on a random tile.
+ * @param rainforest If set the random tile must be in a rainforest zone.
+ */
+static void PlantRandomTree(bool rainforest)
+{
+	uint32_t r = Random();
+	TileIndex tile = RandomTileSeed(r);
+
+	if (rainforest && GetTropicZone(tile) != TROPICZONE_RAINFOREST) return;
+	if (CanPlantTreesOnTile(tile, false)) return;
```

Comment:
```suggestion
	if (!CanPlantTreesOnTile(tile, false)) return;
```

Silly goose, not testing your own code :P